Automattic/wp-calypso
# Onboarding: Remove onboarding vertical step

## Proposed Changes

This PR removes the onboarding vertical screen, and updates the onboarding flow accordingly. In addition, the PR removes the logic that sends the site vertical ID to the theme setup API endpoint.

According to the discussion in p1684827302584859-slack-C03N25JPCE4, the site vertical API endpoint are still being used in the `/setup/ecommerce` flow, independently of the onboarding vertical screen. So the site vertical data layer in Calypso will be kept.

## Testing Instructions



* Head to `/setup`, and expect to land on the Goals screen.
* Ensure that each goal's next step should be the same flow, but without the vertical screen.
* Once you are in the new next step, ensure that the Back button works.

| Goal | Updated next step |
| --- | --- |
| Write & Publish |  `/setup/site-setup/options` |
| Promote myself and business | `/setup/site-setup/designSetup` | 
| Import my existing website content | `/setup/site-setup/import` (no change) |
| Sell online | `/setup/site-setup/options` |
| Get a website built quickly | `/site-setup/difmStartingPoint` (no change) | 
| Other | `/setup/site-setup/designSetup` | 
 
## Pre-merge Checklist
